src/index.tsx

```diff
@@ -11,117 +11,160 @@ import Animated, {
   useSharedValue,
 } from 'react-native-reanimated';
 
+type MarqueeDirection = 'up' | 'left' | 'right' | 'down';
+
 const AnimatedChild = ({
   index,
   children,
   anim,
+  textHeight,
   textWidth,
   spacing,
+  direction = 'right',
 }: React.PropsWithChildren<{
   index: number;
   anim: SharedValue<number>;
+  textHeight: SharedValue<number>;
   textWidth: SharedValue<number>;
   spacing: number;
+  direction?: MarqueeDirection;
 }>) => {
   const stylez = useAnimatedStyle(() => {
+    const vertical = direction === 'up' || direction === 'down';
+    const horizontal = direction === 'left' || direction === 'right';
+
     return {
       position: 'absolute',
-      left: index * (textWidth.value + spacing),
+      top: vertical && direction === 'up' ? index * (textHeight.value + spacing) : undefined,
```

Comment:
you can return split the return for the two cases, instead of having the ternary everywhere.

Like:

```
if (isVertical) {
  return {
    position: 'absolute',
    top: index * (textWidth.value + spacing),
    transform: [{
      translateY: (direction === 'up' ? -1 : 1)(anim.value % (textWidth.value + spacing)),
    }]
  }
} else {
  return {
    position: 'absolute',
    left: index * (textWidth.value + spacing),
    transform: [{
      translateX: (direction === 'left' ? -1 : 1)(anim.value % (textWidth.value + spacing)),
    }]
  }
}
```